web/src/stream_popover.js

```diff
@@ -214,7 +214,29 @@ function build_stream_popover(opts) {
     });
 }
 
-export function build_move_topic_to_stream_popover(
+async function get_message_type(current_stream_id, topic_name, current_message_id) {
+    const lastMessageId = await new Promise((resolve) => {
+        message_edit.with_last_message_id(current_stream_id, topic_name, (message_id) => {
+            resolve(message_id);
+        });
+    });
+
+    const firstMessageId = await new Promise((resolve) => {
+        message_edit.with_first_message_id(current_stream_id, topic_name, (message_id) => {
+            resolve(message_id);
+        });
```

Comment:
We should avoid this structure of doing two API calls, in sequence, when a single would be able to answer our question with `anchor: "newest"` and `before: 1` and `after: 1` -- we'll know we're last if we got 0 after, we'll know we're first if we got 0 before.

Reply:
Fixed by using `anchor: message_id` and `before: 1` and `after: 1` along with the logic you explained above. Thank you for recommending a better way of solving this, made a lot of sense once I realized.